econml/tests/test_drlearner.py

```diff
@@ -398,8 +398,8 @@ def test_drlearner_all_attributes(self):
                                                      0] + np.random.normal(0, sigma, size=(5000,))
         for X in [controls]:
             for W in [None, controls]:
-                for sample_weight in [None, 1 + np.random.randint(10, size=X.shape[0])]:
-                    for sample_var in [None, 1 + np.random.randint(10, size=X.shape[0])]:
+                for sample_weight in [None, np.ones(X.shape[0])]:
```

Comment:
I think we should have some other test where we also test weights that are not all ones and freq_weights that are not all ones and check that we recover the right thing. For instnace, check that with and without summarized we get the same thing, even when we have sample_var and freq_weights and sample_weights, and they are non-ones numbers. 

Such test should exist both for the drlearner and for the dml.

Reply:
Yes, it's under the test_statsmodels file, here it's more to test estimators allow sample_weight argument and a set of estimators allow freq_weight and sample_var as well. There was a bug in this test `if not isinstance(models, StatsModelsLinearRegression)` since `models` is a list, this test is never triggered when sample_var is not None, otherwise it will not pass the test since we can't recover the known coefs with random non zero variance and non ones weight. Under test_statsmodels, I test all the child classes who allow sample_var between summarized and un-summarized data.